scripts/download_checkpoints.py

```diff
@@ -0,0 +1,139 @@
+        return r2_url.replace('r2://olmo-checkpoints/', 'https://olmo-checkpoints.org/')
+    return r2_url
+
+def download_file(url, save_path, chunk_size=8192):
+    """Download a file with progress bar."""
+    response = requests.get(url, stream=True)
+    response.raise_for_status()
+    total_size = int(response.headers.get('content-length', 0))
+    save_path.parent.mkdir(parents=True, exist_ok=True)
+    
+    with open(save_path, 'wb') as f:
+        with tqdm(total=total_size, unit='B', unit_scale=True, desc=save_path.name) as pbar:
+            for chunk in response.iter_content(chunk_size=chunk_size):
+                if chunk:
+                    f.write(chunk)
+                    pbar.update(len(chunk))
+
+def try_get_directory_listing(url):
+    common_files = [
+        "config.yaml",
+        "model.pt",
+        "optim.pt",
+        "train.pt",
+        "model.safetensors",
+        "optim.safetensors",
+    ]
+    
+    found_files = []
+    for pattern in common_files:
+        test_url = urljoin(url.rstrip('/') + '/', pattern)
+        try:
+            response = requests.head(test_url)
+            if response.status_code == 200:
+                found_files.append(pattern)
+        except requests.exceptions.RequestException:
+            continue
+            
+    return found_files
+
+def download_checkpoint(url, save_dir):
+    """Download all files from a checkpoint directory."""
+    r2_url = convert_to_r2_url(url)
+    public_url = convert_to_public_url(r2_url)
+    
+    base_path = Path(save_dir)
+    base_path.mkdir(parents=True, exist_ok=True)
+    
+    print(f"\nR2 URL: {r2_url}")
+    print(f"Public URL: {public_url}")
+    print(f"Saving to: {base_path}")
+    
+    print("Checking for available files...")
+    available_files = try_get_directory_listing(public_url)
+    
+    if not available_files:
+        print("No files found using common patterns. The directory might be empty or use different file patterns.")
+        return
+    
+    for file in available_files:
+        file_url = urljoin(public_url.rstrip('/') + '/', file)
+        file_path = base_path / file
+        
+        try:
+            print(f"\nDownloading: {file}")
+            download_file(file_url, file_path)
+        except requests.exceptions.RequestException as e:
+            print(f"Error downloading {file}: {e}")
+            continue
+
+def main():
+    parser = argparse.ArgumentParser(description='Download OLMo checkpoints from CSV')
+    parser.add_argument('csv_file', type=str, help='Path to the CSV file containing checkpoint URLs')
+    parser.add_argument('--save-dir', type=str, default='./checkpoints',
+                        help='Base directory to save downloaded checkpoints')
+    parser.add_argument('--step', type=str, help='Specific step number to download (optional)')
+    parser.add_argument('--list-steps', action='store_true', help='List available step numbers and exit')
```

Comment:
If you have a tool that can perform multiple different actions, use subcommands.

Reply:
I think Dirk is referring to this: https://docs.python.org/3/library/argparse.html#sub-commands .